### ISSUE TITLE
Dynamic layout / sizing

### ISSUE THREAD

**Issue by hjacobs** [hjacobs/kube-ops-view#6]
Cluster and node sizes need to be completely dynamic. The current node box width/height is hardcoded.

**Comment by hjacobs** [comment]
The strict vertical layout of clusters is a huge waste of screen space as can be seen in this screenshot:
![screenshot_2017-02-13_20-32-21](https://cloud.githubusercontent.com/assets/510328/22899648/d0f20798-f22b-11e6-953e-8d583f3b158f.png)

**Comment by tomislater** [comment]
I am working on it. Here is very early stage (only scaling node height and bars): https://media.giphy.com/media/1BgNwU0jJkeDrQKCrH/giphy.gif

**Comment by tomislater** [comment]
@hjacobs What do you think about it?

* https://media.giphy.com/media/9rewXU06ZFbxcjFfCj/source.mp4
* https://media.giphy.com/media/lcM2I2oQ1HTAwWSMAc/source.mp4
* https://media.giphy.com/media/fHolINczeJZoSkCSnJ/source.mp4

The last one looks somehow crazy, but I added so many pods there ;D

**Comment by hjacobs** [comment]
I don't really like how the different node heights cause the "ragged" look, I would prefer to have something more aligned (but I have no good idea to achieve this).

**Comment by tomislater** [comment]
Yeah, I had no better solution... I have another idea, we can gather information about the largest node (the number of pods is the largest) and use it as a template for all nodes.
For instance, let's say that the largest node has 115 pods, we can get Sqrt of this (10.724) and create a grid 11x11 (now we have six pods per row) for pods in this node (and for every other). It has one drawback; if you had many small nodes (with a small number of pods) it would look somehow strange.

**Comment by hjacobs** [comment]
Let's try it out..

**Comment by tomislater** [comment]
Looks much better, methinks:
<img width="833" alt="screenshot 2019-03-02 at 14 30 32" src="https://user-images.githubusercontent.com/1127164/53682619-da62af80-3cf7-11e9-97ab-7eb22061baab.png">
<img width="626" alt="screenshot 2019-03-02 at 14 29 27" src="https://user-images.githubusercontent.com/1127164/53682620-dafb4600-3cf7-11e9-9ea0-f84cc081941c.png">
<img width="627" alt="screenshot 2019-03-02 at 14 28 25" src="https://user-images.githubusercontent.com/1127164/53682621-dafb4600-3cf7-11e9-86d8-e05cf92e4724.png">
<img width="582" alt="screenshot 2019-03-02 at 14 28 00" src="https://user-images.githubusercontent.com/1127164/53682622-dafb4600-3cf7-11e9-874e-8091768ba236.png">

Nodes on the second screen looks strange... so we could keep minimal width/height.

**Comment by tomislater** [comment]
<img width="683" alt="screenshot 2019-03-02 at 14 36 46" src="https://user-images.githubusercontent.com/1127164/53682720-a2a83780-3cf8-11e9-9384-2c1df186aad4.png">

I have improved the gap between kube-system pods and others.

**Comment by hjacobs** [comment]
:+1: Looks better than the variable height and if we keep the minimum size as before the layout would be the same for people with small notes.

**Comment by tomislater** [comment]
Another improvement: #217

**Comment by XenGi** [comment]
The current version looks really good. What I'm missing is the ability to change the rectangle of the cluster. ATM it's always in a 4:3 scale but my monitor is 16:9 so there's always some space wasted. Can you detect what the screen ratio is and adapt the rectangle or give the user the ability to scale the rectangle?

**Comment by guillermodep** [comment]
Hi! I need to resize the layout, could you please tell me how to add more columns, is an environment or what? thaks
